### ISSUE TITLE
Curly braces in encapsed string are misparsed

### ISSUE THREAD

**Issue by pornel** [glayzzle/php-parser#73]
```php
"${bar}"
```

is parsed as

```js
[ Variable {
    kind: 'variable',
    name:
     Variable {
       kind: 'variable',
       name: 'bar',
```

However, this is equivalent to just `$bar` in PHP, and parsed `name` should just be a string, not a variable-variable.

Test:

```sh
php -r '$single="double"; $bar="single"; echo "${bar}\n";'
```
prints "single".

```sh
php -r '$single="double"; $bar="single"; echo "${$bar}\n";'
```

prints "double", but both are parsed the same by this parser.

**Comment by ichiriac** [comment]
Related to another issue : https://github.com/glayzzle/php-parser/issues/59
---
The decorative curlies around the variable are used to limit the variable name around identifier like text. For exemple : `echo "Hello ${foo}bar";`

Actually the variable name can be : 
 * String
 * Variable
 * ConstRef (a constant variable name)
 * OffsetLookup
 * Expr

If in this case I use a String, the curlies information is loosed, so it will be hard to rebuild a PHP source from AST (for example [php-unparser](https://github.com/glayzzle/php-unparser/blob/migrate-php-parser-2.0.0/node_translators/variable.js#L5)) - I will add a new curly property on variable node.

The `echo "${bar}"` should be :

```
Variable {
    kind: 'variable',
    curly: true,
    name: 'bar',
```

**Comment by ichiriac** [comment]
Hi @pornel,

Thanks for the report, it is fixed and released under 2.0.4 / update your npm version

**Comment by pornel** [comment]
Works. Thank you.
